src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -181,6 +183,39 @@ internal async Task<(bool, SumType<CompletionItem[], CompletionList>?)> TryGetPr
             return (true, result);
         }
 
+        private SumType<CompletionItem[], CompletionList>? IncludeCSharpKeywords(SumType<CompletionItem[], CompletionList> completionResult, LanguageServerKind kind)
+        {
+            var result = completionResult.Match<SumType<CompletionItem[], CompletionList>?>(
+                items =>
+                {
+                    var newList = items.Union(_keywordCompletionItems, CompletionComparer.Instance);
+                    return newList.ToArray();
+                },
+                list =>
+                {
+                    var newList = list.Items.Union(_keywordCompletionItems, CompletionComparer.Instance);
+                    list.Items = newList.ToArray();
+
+                    return list;
+                });
+
+            return result;
+        }
+
+        private static readonly IReadOnlyCollection<string> _keywords = new string[] {
+            "for", "foreach", "while", "switch", "lock",
+            "case", "if", "try", "do", "using"
+        };
+
+        private static readonly IReadOnlyCollection<CompletionItem> _keywordCompletionItems = _keywords.Select(k => new CompletionItem {
+                Label = k,
+                InsertText = k,
+                FilterText = k,
+                Kind = CompletionItemKind.Keyword,
+                SortText = k,
+                InsertTextFormat = InsertTextFormat.Plaintext,
+            }).ToList();
```

Comment:
Nit;
```suggestion
            }).ToArray();
```

though given you've specified `IReadOnlyCollection` I wonder if the compiler handles that?